src/components/Survey/Nickname/Nickname.tsx

```diff
@@ -12,60 +11,85 @@ const {
   s_title,
   s_errorMsg,
   s_errorIcon,
-} = styles;
+  s_nextButton,
+} = style;
 
 interface Props {
   nickname: string;
   setNickname: (value: string) => void;
   setNextPage: () => void;
 }
 
-function Nickname({ nickname, setNickname, setNextPage }: Props) {
-  const [isDisabled, setIsDisabled] = useState<boolean>(!nickname);
+const Nickname = ({ nickname, setNickname, setNextPage }: Props) => {
   const [errorMessage, setErrorMessage] = useState<string>('');
+  const isDisabled = useMemo(() => !nickname || !!errorMessage, [nickname, errorMessage]);
 
-  const checkNicknameValidation = (value: string) => {
-    const nicknameRegex = /^[가-힣\s|ㄱ-ㅎ|a-z|A-Z|0-9|_|.|,]+$/g;
-    const nicknameLengthRegex = /^.{2,8}$/g;
+  useEffect(() => {
+    const isValidNickname = (userNickname: string) => {
+      const nicknameSpaceRegex = /\s/g;
+      const specialCharactersRegex = /[`~!@#$%^&*()|+\-=?;:'"<>\\{\\}\\[\]\\\\/]/g;
+      const nicknameRegex = /^[가-힣\s|ㄱ-ㅎ|a-z|A-Z|0-9|_|.|,]+$/g;
+      const nicknameLengthRegex = /^.{2,8}$/g;
 
-    if (!nicknameRegex.test(value)) {
-      setErrorMessage('한글, 영어, 숫자, 특수문자(_.,)만 가능해요');
-      setIsDisabled(true);
-    } else if (!nicknameLengthRegex.test(value)) {
-      setErrorMessage('2 ~ 8글자만 가능해요');
-      setIsDisabled(true);
+      if (nicknameSpaceRegex.test(userNickname)) {
+        return '공백을 제거해 주세요.';
+      }
+      if (specialCharactersRegex.test(nickname)) {
+        return '특수문자는 . , _ 만 가능해요';
+      }
+
+      if (!nicknameRegex.test(userNickname)) {
+        return '한글, 영어, 숫자, 특수문자(_.,)만 가능해요';
+      }
+      if (!nicknameLengthRegex.test(userNickname)) {
+        return '2 ~ 8글자만 가능해요';
+      }
+
+      return null;
+    };
+
+    const nickNameError = isValidNickname(nickname);
+
+    if (nickNameError) {
+      setErrorMessage(nickNameError);
     } else {
       setErrorMessage('');
-      setIsDisabled(false);
     }
-  };
+  }, [nickname]);
 
-  // onChange
   const handleChangeNickname = (e: React.ChangeEvent<HTMLInputElement>): void => {
     const { value }: { value: string } = e.target;
     setNickname(value);
-    checkNicknameValidation(value);
   };
 
   return (
     <section className={classNames(s_container)}>
-      <h2 className={classNames(s_title)}>당신을 뭐라고 불러드릴까요?</h2>
+      <h2 className={classNames(s_title)}>
+        <p>당신을</p> 뭐라고 불러드릴까요?
+      </h2>
```

Comment:
제목 요소가 페이지에서 가장 크게 보이는 텍스트는 아니라는 생각이 들어요~~
시멘틱한 마크업의 관점에서 h2는 제목으로 생각되고 쓰는것인데 '당신을 뭐라고 불러드릴까요?' 라는 제목은 조금 어색한거 같습니다. UI적으로는 숨김요소로 처리하되 제목을 '닉네임 입력' 으로 하는것이 더 괜찮아 보이는데 어떠신가요?

결국은 아래와 같이 될 것 같습니다.
```suggestion
      <h2 className={classNames('s_a11yHidden')}>닉네임 입력</h2>
      <p className={classNames(s_title)}>
        <span className={classNames('s_whiteSpace')}>당신을</span> 뭐라고 불러드릴까요?
      </p>
```